src/parser/parse_tree_cl.c

```diff
@@ -18260,16 +18261,52 @@ pt_is_nested_expr (const PT_NODE * node)
 }
 
 /*
- *   pt_is_allowed_as_function_index () : checks if the given operator
- *					  is allowed in the structure of a
- *					  function index
+ *   pt_function_is_allowed_as_function_index () : checks if the given function
+ *						   is NOT allowed in the structure of a
+ *						   function index
+ *   return:
+ *   func(in): parse tree node function
+ */
+static bool
+pt_function_is_allowed_as_function_index (const PT_NODE * func)
+{
+  assert (func != NULL && func->node_type == PT_FUNCTION);
+
+  // TODO: expose get_signatures () of func_type.cpp & filter out funcs returning PT_TYPE_JSON
+  switch (func->info.function.function_type)
+    {
+      // TODO: investigate whether there are other functions that are treated underneath a PT_FUNCTION_HOLDER
```

Comment:
I think we may remove the TODO, if we explicitly add functions codes which are not allowed on true case or as comments (under true case).

The following are missing, I added how I think we should handle them:
F_INSERT_SUBSTRING, F_ELT - legacy functions, should be allowed

 F_JSON_CONTAINS, F_JSON_CONTAINS_PATH, F_JSON_DEPTH, F_JSON_LENGTH, F_JSON_TYPE, F_JSON_VALID - json, should be allowed

F_JSON_PRETTY, F_JSON_QUOTE, F_JSON_UNQUOTE- json, should not be allowed

F_BENCHMARK  - should be allowed